lib/services/s3.js

```diff
@@ -772,7 +772,18 @@ AWS.util.update(AWS.S3.prototype, {
     var expires = params.Expires || 900;
     delete params.Expires; // we can't validate this
     var request = this.makeRequest(operation, params);
-    return request.presign(expires, callback);
+    var hasNextTick = typeof process === 'object' && typeof process.nextTick === 'function';
```

Comment:
Could this check for and use `setImmediate` instead of `process.nextTick`? `setImmediate` is also supported in IE and Edge.